EIPS/eip-7498.md

```diff
@@ -0,0 +1,299 @@
+---
+eip: 7498
+title: NFT Redeemables
+description: Extension to ERC-721 and ERC-1155 for onchain and offchain redeemables
+author: Ryan Ghods (@ryanio), 0age (@0age), Adam Montgomery (@montasaurus)
+discussions-to: https://ethereum-magicians.org/t/erc-7498-nft-redeemables/15485
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-07-28
+requires: 165, 712, 721, 1155, 1271
+---
+
+## Abstract
+
+This specification introduces a new interface that extends [ERC-721](./eip-721.md) and [ERC-1155](./eip-1155.md) to enable the discovery and use of onchain and offchain redeemables for NFTs.
+
+## Motivation
+
+Creators frequently use NFTs to create redeemable entitlements for digital and physical goods. However, without a standard interface, it is challenging for users and apps to discover and interact with these NFTs in a predictable and standard way. This standard aims to encompass enabling broad functionality for:
+
+- discovery: events and getters that provide information about the requirements of a redemption campaign
+- onchain: token mints with context of items spent
+- offchain: the ability to associate with ecommerce orders (through `redemptionHash`)
+- trait redemptions: improving the burn-to-redeem experience with E RC-7496 <!-- [E RC-7496] (./e ip-7496.md) --> Dynamic Traits.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+The token MUST have the following interface and MUST return `true` for [ERC-165](./eip-165.md) supportsInterface for `0x12345678`, the 4 byte interfaceId of the below.
+
+```solidity
+interface IERC7501 {
+  /* Events */
+  event CampaignUpdated(uint256 indexed campaignId, CampaignParams params, string URI);
```

Comment:
suggestion: change "event CampaignUpdated(uint256 indexed campaignId, CampaignParams params, string URI);" to "event CampaignUpdated(uint256 campaignId, CampaignParams params, string URI); for saving gas"

Reply:
having the indexed param is useful for querying eth_getLogs based on the campaignId, but will consider removing. what do others think?

Reply:
in discussion with the rest of my team, although there is a little gas savings we think it's worth it to keep indexed on campaignId